lib/rules/semi-spacing.js

```diff
@@ -104,22 +104,52 @@ module.exports = function(context) {
 
             if (hasLeadingSpace(token)) {
                 if (!requireSpaceBefore) {
-                    context.report(node, location, "Unexpected whitespace before semicolon.");
+                    context.report({
```

Comment:
can we have a central function (like report) which handles the reporting? 
So that we dont have to keep changing this at 4 spots.